internal/resources/app_endpoint_oidc.go

```diff
@@ -0,0 +1,403 @@
+) {
+	// The import ID should be in the format: organizationId,projectId,clusterId,appServiceId,app_endpoint_name,provider_id
+	resource.ImportStatePassthroughID(ctx, path.Root("app_endpoint_name"), req, resp)
+}
+
+func (r *AppEndpointOidcProvider) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	data, ok := req.ProviderData.(*providerschema.Data)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *ProviderSourceData, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.Data = data
+}
+
+// Create creates a new OIDC provider and stores provider_id.
+func (r *AppEndpointOidcProvider) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan providerschema.AppEndpointOidcProvider
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	organizationId := plan.OrganizationId.ValueString()
+	projectId := plan.ProjectId.ValueString()
+	clusterId := plan.ClusterId.ValueString()
+	appServiceId := plan.AppServiceId.ValueString()
+	appEndpointName := plan.AppEndpointName.ValueString()
+
+	url := fmt.Sprintf(
+		"%s/v4/organizations/%s/projects/%s/clusters/%s/appservices/%s/appEndpoints/%s/oidcProviders",
+		r.HostURL,
+		organizationId,
+		projectId,
+		clusterId,
+		appServiceId,
+		appEndpointName,
+	)
+
+	payload := buildAppEndpointOIDCProviderPayload(plan)
+
+	cfg := api.EndpointCfg{Url: url, Method: http.MethodPost, SuccessStatus: http.StatusCreated}
+	res, err := r.Client.ExecuteWithRetry(ctx, cfg, payload, r.Token, map[string]string{"Content-Type": "application/json"})
+	if err != nil {
+		resp.Diagnostics.AddError("Error Creating OIDC Provider", api.ParseError(err))
+		return
+	}
+
+	// Capture providerId from response
+	var created api.AppEndpointOIDCProviderResponse
+	if err := json.Unmarshal(res.Body, &created); err == nil {
+		if created.ProviderID != "" {
+			plan.ProviderId = types.StringValue(created.ProviderID)
+		}
+	}
+
+	// Initialize optional/computed attributes to null before refresh to preserve user intent
+	initOidcProviderNullsBeforeRefresh(&plan)
+	diags = resp.State.Set(ctx, plan)
+	resp.Diagnostics.Append(diags...)
+
+	// Refresh using GET; preserve nulls for optional attributes during create
+	if !plan.ProviderId.IsNull() && !plan.ProviderId.IsUnknown() {
+		details, err := r.getOidcProvider(ctx, organizationId, projectId, clusterId, appServiceId, appEndpointName, plan.ProviderId.ValueString())
+		if err == nil {
+			r.mapResponseToState(&plan, details, true)
+		} else {
+			tflog.Warn(ctx, "Failed to read OIDC provider after creation", map[string]any{"error": api.ParseError(err)})
```

Comment:
no need to check for nil

no need to log warning

it is necessary to return a warning

do something like:

```
details, err := r.getOidcProvider(ctx, organizationId, projectId, clusterId, appServiceId, appEndpointName, plan.ProviderId.ValueString())
		if err != nil {
			resp.Diagnostics.AddError(
			"Error Reading Capella Collection",
			"Could not read Capella Collection for the scope: %s "+scopeName+"."+errorMessageAfterCollectionCreation+collection_api.ParseError(err),
		)
		return
		}

r.mapResponseToState(&plan, details, true)

```